### PR TITLE
Update the statistic info of hash join temporary files

### PR DESCRIPTION
**Pull request by yaowangm**
1. Remove the max/min file size in the statistic info of temporary files, and add the average file size.
2. Include the statistic info of temporary files of the hash table and the outer table.

### CODE REVIEW

**Comment by adam8157** [comment]
I would suggest having a subject describing what it does, instead of merging codes.

**Review by adam8157 — APPROVED** [review]

**Comment by adam8157** [comment]
I updated the subject and message since it's not related to the GUC actually, and it doesn't have that test case change.

**Comment by yaowangm** [comment]
@adam8157 We must have the test change, since we changed the output of avg file size and the calculation of compression buffer size, which changed the behavior of disabling compression. Otherwise we will break the pipeline. I have updated the test case, please take another look. Thanks!

**Comment by yaowangm** [comment]
@adam8157 As talked, I made a minor change to avoid breaking ABI. Please take another look, thanks!

File: src/include/executor/hashjoin.h [hunk 1]
```diff
@@ -409,7 +409,8 @@ typedef struct HashJoinTableData
 	/* Statistic info of work file set, copied from work_set */
 	uint32		workset_num_files;
 	uint32		workset_num_files_compressed;
-	uint64		workset_max_file_size;
+	uint64		workset_avg_file_size;
+	/* Not used, just to for ABI compatibility */
 	uint64		workset_min_file_size;
```
Comment (adam8157) [inline src/include/executor/hashjoin.h:414]: Just rename it to `workset_abi_reserved`.

**Review by adam8157 — APPROVED** [review]
LGTM, but please rename the not-used variable too, IDE would ignore the comment and suggest it anyway, renaming it will help others.

File: src/include/executor/hashjoin.h (hunk 1 above)
Comment (yaowangm) [reply src/include/executor/hashjoin.h]: Done.

**Pull request merged**